### PR TITLE
add notes for 28122

### PR DESCRIPTION
**Pull request by josibake**

### CODE REVIEW

File: _posts/2023-08-02-#28122.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+## Questions
+
+1. Did you read the [BIP](https://github.com/bitcoin/bips/pull/1458)? In a few sentences, can you summarize it in your own words?
+
+1. Did you review the [reference implementation](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/reference.py) and [test vectors](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/send_and_receive_test_vectors.json)? Can you think of additional test cases?
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
+
+1. In [5688262](https://github.com/bitcoin/bitcoin/pull/28122/commits/56882622faf469b6f948f79a69c3c8ddbde92ff8), why do we need a custom hash function? Why not just use the default hash function from `libsecp256k1`? (hint: [see this discussion on the BIP](https://github.com/bitcoin/bips/pull/1458#pullrequestreview-1466163601))
```
Comment (glozow) [inline _posts/2023-08-02-#28122.md:27]: For these and the other links, we'll want to use the review club mirror, otherwise they'll disappear as soon as you force push to the PR. I've pushed a tag at https://github.com/bitcoin-core-review-club/bitcoin/commits/pr28122

```suggestion
1. In [5688262](https://github.com/bitcoin-core-review-club/bitcoin/commit/56882622faf469b6f948f79a69c3c8ddbde92ff8), why do we need a custom hash function? Why not just use the default hash function from `libsecp256k1`? (hint: [see this discussion on the BIP](https://github.com/bitcoin/bips/pull/1458#pullrequestreview-1466163601))
```

File: _posts/2023-08-02-#28122.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+1. [BIP352](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352.mediawiki#versions) talks about about versioning and forward compatibility. Why is this important? Do you see any issues with the way the PR is written with respect to forwards compatibility?
+
+1. In [`DecodeSilentAddress`](https://github.com/bitcoin/bitcoin/pull/28122/commits/ae6019b3a5f99e63b5f23e48fd430147b8500877#diff-a83ac7cb4a9d60b647db6378ccfad815005724e594bf53f49745b0f40701292fR327) there is a check on the version and data size. In your own words, can you explain what this check is doing? Why is it important?
+
+1. The bulk of the silent payments code is under the wallet directory in [src/wallet/silentpayments.cpp](https://github.com/bitcoin/bitcoin/blob/2dedccebac504b6f3aea68cea6eb7537a16fd8ea/src/wallet/silentpayments.cpp). Is this a good place for the code to live? Why or why not? Can you think of use case where it would be better to have the code outside the wallet?
```
Comment (glozow) [inline _posts/2023-08-02-#28122.md:35]: Same here, use the mirror repo to avoid the links going dead

File: _posts/2023-08-02-#28122.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+1. Based on your answer above, what's one benefit for how the `Recipient` class is currently written?
+
+1. In a few sentences, can you summarize what [`Recipient::ScanTxOutputs`](https://github.com/bitcoin/bitcoin/pull/28122/commits/2dedccebac504b6f3aea68cea6eb7537a16fd8ea#diff-44259d668fd117234d32cc93447c309aaeabcc780fdd38b1de6a187484040f8cR50) is doing?
+
+1. In [`Sender::CreateOutput`](https://github.com/bitcoin/bitcoin/pull/28122/commits/2dedccebac504b6f3aea68cea6eb7537a16fd8ea#diff-44259d668fd117234d32cc93447c309aaeabcc780fdd38b1de6a187484040f8cR94), we hash the ECDH shared secret with an integer. Why do we need the integer counter?
+
+1. Bitcoin Core has a [HashWriter](https://github.com/josibake/bitcoin/blob/2dedccebac504b6f3aea68cea6eb7537a16fd8ea/src/hash.h) class, which we use. Can you describe what this class does? For our usecase, we need to hash a public key (the shared secret) and an integer. The HashWriter provides a template function for hashing two objects: why not use this? Even better, why not hash the pubkey with the `<<` operator? (hint: look at how a `CPubKey` is serialized)
```
Comment (glozow) [inline _posts/2023-08-02-#28122.md:45]: nit
```suggestion
1. Bitcoin Core has a [`HashWriter`](https://github.com/josibake/bitcoin/blob/2dedccebac504b6f3aea68cea6eb7537a16fd8ea/src/hash.h) class, which we use. Can you describe what this class does? For our usecase, we need to hash a public key (the shared secret) and an integer. The HashWriter provides a template function for hashing two objects: why not use this? Even better, why not hash the pubkey with the `<<` operator? (hint: look at how a `CPubKey` is serialized)
```

File: _posts/2023-08-02-#28122.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+1. In [5688262](https://github.com/bitcoin/bitcoin/pull/28122/commits/56882622faf469b6f948f79a69c3c8ddbde92ff8), why do we need a custom hash function? Why not just use the default hash function from `libsecp256k1`? (hint: [see this discussion on the BIP](https://github.com/bitcoin/bips/pull/1458#pullrequestreview-1466163601))
+
+1. [ae6019b](https://github.com/bitcoin/bitcoin/pull/28122/commits/ae6019b3a5f99e63b5f23e48fd430147b8500877) and [439e57a](https://github.com/bitcoin/bitcoin/pull/28122/commits/439e57aa88d33e9c61e4c38a37526133a79ab188) add functions for decoding and encoding silent payment addresses, but [src/key_io.cpp](https://github.com/bitcoin/bitcoin/blob/439e57aa88d33e9c61e4c38a37526133a79ab188/src/key_io.cpp) already has a `DestinationEncoder` class and a `DecodeDestination` function. Why can't we just add silent payment addresses to the existing encoder and decoder? More specifically, what is a `CTxDestination`? How is it different from a silent payment address? (hint: `CTxDestination` is defined in [src/script/standards.h](https://github.com/josibake/bitcoin/blob/439e57aa88d33e9c61e4c38a37526133a79ab188/src/script/standard.h))
```
Comment (glozow) [inline _posts/2023-08-02-#28122.md:29]: I had difficulty understanding what was being asked, probably best to split this up into multiple questions or add the background information as notes

File: _posts/2023-08-02-#28122.md (hunk 1 above)
Comment (josibake) [reply _posts/2023-08-02-#28122.md:27]: :+1: Is it possible to do the same for the BIPs repo? There are links to the reference.py implementation and test vectors file from the BIP pull request, those will also break once if I need to update the BIP. Alternatively, I can just have all the BIP links point just to the PR, whichever is easier.

Comment (glozow) [reply _posts/2023-08-02-#28122.md:27]: Ah yes no problem, I made a mirror and pushed a tag: https://github.com/bitcoin-core-review-club/bips/tree/pr28122

Feel free to use e.g. https://github.com/bitcoin-core-review-club/bips/blob/174ba63f44ddab89b73c21f90d2f5cb1968b6db1/bip-0352/bech32m.py

File: _posts/2023-08-02-#28122.md (hunk 4 above)
Comment (josibake) [reply _posts/2023-08-02-#28122.md]: Added some notes to the background section and tried to rephrase this to be a little more clear

File: _posts/2023-08-02-#28122.md (hunk 3 above)
Comment (josibake) [reply _posts/2023-08-02-#28122.md]: Fixed

File: _posts/2023-08-02-#28122.md (hunk 2 above)
Comment (josibake) [reply _posts/2023-08-02-#28122.md]: Fixed

File: _posts/2023-08-02-#28122.md (hunk 1 above)
Comment (josibake) [reply _posts/2023-08-02-#28122.md]: Done!

**Comment by glozow** [comment]
Thanks @josibake, looking forward to the meetings!
